### PR TITLE
Update poetry ci

### PR DESCRIPTION
**Pull request by uSpike**

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -45,6 +45,46 @@ jobs:
...
           poetry install --no-interaction
+
+      - name: Run tests
+        run: poetry run pytest -v
+  lint:
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["3.11"]
+    runs-on: ubuntu-18.04
```
Comment (Mindstorman) [inline .github/workflows/main.yml:56]: Probably want to update to a newer ubuntu release as this is EOL by April 30th, 2023

https://github.blog/changelog/2022-08-09-github-actions-the-ubuntu-18-04-actions-runner-image-is-being-deprecated-and-will-be-removed-by-12-1-22/

Comment (uSpike) [reply .github/workflows/main.yml:56]: Mind if I make a separate issue for that?

Comment (uSpike) [reply .github/workflows/main.yml:56]: https://github.com/uSpike/rpcx/issues/70

Comment (Mindstorman) [reply .github/workflows/main.yml:56]: Sure that works.

**Review by Mindstorman — APPROVED** [review]
Looks good. #70 will address the concerns with Ubuntu 18.04 going EOL

**Pull request merged**